client/service/src/chain_ops.rs

```diff
@@ -175,25 +343,15 @@ impl<
 					"Stopping after #{} blocks because of an error",
 					link.imported_blocks,
 				);
+				// We've encountered an error, we can stop here.
 				return std::task::Poll::Ready(Ok(()));
```

Comment:
Should be also an error.

Reply:
This was exactly what I was asking in [a previous comment](https://github.com/paritytech/substrate/pull/5816#issuecomment-621975433) :p